packages/ibm-products/src/components/Checklist/Checklist.js

```diff
@@ -0,0 +1,316 @@
+
+import { ChecklistIcon } from './ChecklistIcon';
+import { ChecklistChart } from './ChecklistChart';
+
+// The block part of our conventional BEM class names (blockClass__E--M).
+const blockClass = `${pkg.prefix}--checklist`;
+const componentName = 'Checklist';
+
+// NOTE: the component SCSS is not imported here: it is rolled up separately.
+
+// Default values can be included here and then assigned to the prop params,
+// e.g. prop = defaults.prop,
+// This gathers default values together neatly and ensures non-primitive
+// values are initialized early to avoid react making unnecessary re-renders.
+// Note that default values are not required for props that are 'required',
+// nor for props where the component can apply undefined values reasonably.
+// Default values should be provided when the component needs to make a choice
+// or assumption when a prop is not supplied.
+
+// Default values for props
+const defaults = {
+  onClickViewAll: () => {},
+  onToggle: () => {},
+  open: true,
+  showToggle: true,
+  taskLists: [],
+  theme: 'light',
+};
+
+/**
+ * TODO: A description of the component.
+ */
+export let Checklist = React.forwardRef(
+  (
+    {
+      className,
+      onClickViewAll = defaults.onClickViewAll,
+      onToggle = defaults.onToggle,
+      open = defaults.open,
+      chartValue,
+      chartLabel,
+      showToggle = defaults.showToggle,
+      taskLists = defaults.taskLists,
+      theme = defaults.theme,
+      title,
+      viewAllLabel,
+      ...rest
+    },
+    ref
+  ) => {
+    const [isOpen, setIsOpen] = useState(open);
+
+    // Don't use this test: {chartValue && chartLabel && (render...)}.
+    // If `chartValue` is 0 (zero) - a valid value - then it will render 0 and skip the remaining statement.
+    // Use this test instead: {chartLabelAndValue && (render...)}.
+    // The ChecklistChart component will validate `chartValue`.
+    const chartLabelAndValue = typeof chartValue === 'number' && chartLabel;
+
+    const handleClickToggle = () => {
+      setIsOpen((prevOpen) => !prevOpen);
+    };
+
+    const handleClickViewAll = () => {
+      onClickViewAll();
+    };
+
+    // If state changes, then trigger callback.
+    useEffect(() => {
+      onToggle(isOpen);
+    }, [isOpen, onToggle]);
+
+    // If the "open" prop is changed after initialization,
+    // then update internal state.
+    useEffect(() => {
+      setIsOpen(open);
+    }, [open]);
+
+    return (
+      <aside
+        {
+          // Pass through any other property values as HTML attributes.
+          ...rest
+        }
+        className={cx(
+          blockClass, // Apply the block class to the main HTML element
+          className, // Apply any supplied class names to the main HTML element.
+          // example: `${blockClass}__template-string-class-${kind}-n-${size}`,
+          {
+            [`${blockClass}__closed`]: !isOpen,
+          }
+        )}
+        ref={ref}
+        role="main"
```

Comment:
I think this is left from the defaults, but think we can remove this role since `aside` should already have an implicit role.

Reply:
Removed.